terraform/environments/data-platform/iam.tf

```diff
@@ -420,6 +420,62 @@ data "aws_iam_policy_document" "logs_s3_bucket_policy_document" {
     ]
   }
 
+  statement {
+    sid    = "AllowPutFromCloudtrail"
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:PutObject"
+    ]
+
+    resources = [
+      "${module.logs_s3_bucket.bucket.arn}/AWSLogs/${data.aws_caller_identity.current.account_id}/*"
+    ]
+    
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+
+      values = [
+        "bucket-owner-full-control"
+      ]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+
+      values   = [aws_cloudtrail.data_s3_put_objects.arn]
```

Comment:
Maybe you could break the dependency cycle by hardcoding the value here, if the arn is predictable. Or maybe the policy can be split into one that depends on the cloudtrail and one that is depended on by it. (I'm not sure exactly what is meant to be dependant on what)